core/src/main/java/org/apache/spark/shuffle/sort/UnsafeShuffleWriter.java

```diff
@@ -360,12 +368,10 @@ void forceSorterToSpill() throws IOException {
 
     final OutputStream bos = new BufferedOutputStream(
             new FileOutputStream(outputFile),
-            (int) sparkConf.getSizeAsKb("spark.shuffle.unsafe.file.output.buffer", "32k") * 1024);
+            outputBufferSizeInBytes);
```

Comment:
Just to understand what is happening.

1. Shuffle records are written to a serialisation buffer (1M) after serialisation
2. The serialised buffer is written to in-memory-sorter’s buffer
3. once in-memory sorter’s buffer is full, the data is copied to sorter’s disk buffer (1M)
4. the sorter’s disk buffer is written out to a buffered output stream (buffer = 32k)

I am guessing reducing the sorter’s disk buffer (in step 3) is helping because it triggers fewer writes/allocations in a single call at step 4 (and allowing more parallelism in writing back to disk and copying of data).